Rubberduck.VBEditor.VB6/SafeComWrappers/VB/VBControl.cs

```diff
@@ -11,10 +12,54 @@ public VBControl(VB.VBControl target, bool rewrapping = false)
         {
         }
 
+        public IProperties Properties => new Properties(IsWrappingNullReference? null : Target.Properties);
+        
         public string Name
         {
-            get => IsWrappingNullReference ? string.Empty : Target.Properties.Item("Name").Value.ToString();
-            set { if (!IsWrappingNullReference) Target.Properties.Item("Name").Value = value; }
+            get
+            {
+                if (IsWrappingNullReference)
+                {
+                    return string.Empty;
+                }
+
+                using (var properties = this.Properties)
+                {
+                    foreach (var property in properties)
+                    {
+                        using (property)
+                        {
+                            if (property.Name == "Name")
```

Comment:
@MDoerner is it necessary to do a `foreach` to locate a specific thing here? I think it should not leak if one does  this way:
```
using (var properties = this.Properties;
using (var property = properties.Item("Name");
{
  return property.value as string;
}
```

Reply:
This seems reasonable, but it will require a try catch block because the properties collection will throw when asked for a non-existent key.